### PR TITLE
refactor: Removed dependency on lodash.forOwn

### PR DESCRIPTION
**Pull request by zashraf1985**
## Summary
Removed `lodash.forOwn` function and replaced all the usaes with `Object.keys`. This is part of an effort to remove lodash dependency to reduce bundle size.

## Test Plan
All Unit tests and Full Stack SDK compatibility passed

### CODE REVIEW

File: packages/optimizely-sdk/lib/utils/fns/index.js [hunk 1]
```diff
@@ -32,7 +32,14 @@ module.exports = {
...
   forEach: require('lodash/forEach'),
-  forOwn: require('lodash/forOwn'),
+  forOwn: function(obj, callback) {
+    for(var key in obj) {
+      if(obj.hasOwnProperty(key)){
+        var val = obj[key];
+        callback(val, key, obj);
+      }
+    }
+  },
```
Comment (mjc1283) [inline packages/optimizely-sdk/lib/utils/fns/index.js]: This looks good 👍 but could you double check all the places we're using this?

lodash `forOwn` is more complex, for example:
- If you pass `null` or `undefined` as the first argument it won't crash
- If the callback returns `false` it breaks out of the loop early

There could be other differences as well. Let's just check all the places we're using it and make sure we are not relying on any of this kind of special functionality.

Comment (zashraf1985) [reply packages/optimizely-sdk/lib/utils/fns/index.js]: early return feature is used in some places in the code. I added that functionality and added unit tests.

File: packages/optimizely-sdk/lib/core/notification_center/index.js [hunk 2]
```diff
@@ -101,19 +103,18 @@ NotificationCenter.prototype.removeNotificationListener = function (listenerId)
...
           indexToRemove = i;
           typeToRemove = notificationType;
           return false;
         }
       });
-      if (indexToRemove !== undefined && typeToRemove !== undefined) {
-        return false;
-      }
-    });
-
```
Comment (mjc1283) [inline packages/optimizely-sdk/lib/core/notification_center/index.js]: Using `Array.prototype.forEach`, we no longer exit early from the loop when `false` is returned.  I suggest `Array.prototype.some` to maintain the existing behavior.

File: packages/optimizely-sdk/lib/core/notification_center/index.js [hunk 3]
```diff
@@ -130,7 +131,8 @@ NotificationCenter.prototype.removeNotificationListener = function (listenerId)
  */
 NotificationCenter.prototype.clearAllNotificationListeners = function () {
   try{
-    fns.forOwn(enums.NOTIFICATION_TYPES, function (notificationTypeEnum) {
+    Object.keys(enums.NOTIFICATION_TYPES).forEach(function (key) {
```
Comment (mjc1283) [inline packages/optimizely-sdk/lib/core/notification_center/index.js]:
```suggestion
    jsSdkUtils.objectValues(enums.NOTIFICATION_TYPES).forEach(function (notificationTypeEnum) {
```

File: packages/optimizely-sdk/lib/core/project_config/index.js [hunk 4]
```diff
@@ -60,7 +60,8 @@ module.exports = {
     });
 
     projectConfig.rolloutIdMap = fns.keyBy(projectConfig.rollouts || [], 'id');
-    fns.forOwn(projectConfig.rolloutIdMap, function(rollout) {
+    Object.keys(projectConfig.rolloutIdMap || {}).forEach(function (rolloutId) {
```
Comment (mjc1283) [inline packages/optimizely-sdk/lib/core/project_config/index.js]:
```suggestion
    jsSdkUtils.objectValues(projectConfig.rolloutIdMap || {}).forEach(function (rollout) {
```

File: packages/optimizely-sdk/lib/core/project_config/index.js [hunk 5]
```diff
@@ -79,8 +80,8 @@ module.exports = {
 
       // Creates { <variationId>: { key: <variationKey>, id: <variationId> } } mapping for quick lookup
       fns.assignIn(projectConfig.variationIdMap, fns.keyBy(experiment.variations, 'id'));
-
-      fns.forOwn(experiment.variationKeyMap, function(variation) {
+      Object.keys(experiment.variationKeyMap || {}).forEach(function (variationKey) {
```
Comment (mjc1283) [inline packages/optimizely-sdk/lib/core/project_config/index.js]:
```suggestion
      jsSdkUtils.objectValues(experiment.variationKeyMap || {}).forEach(function (variation) {
```

File: packages/optimizely-sdk/lib/core/project_config/index.js [hunk 6]
```diff
@@ -92,7 +93,8 @@ module.exports = {
     projectConfig.experimentFeatureMap = {};
 
     projectConfig.featureKeyMap = fns.keyBy(projectConfig.featureFlags || [], 'key');
-    fns.forOwn(projectConfig.featureKeyMap, function(feature) {
+    Object.keys(projectConfig.featureKeyMap || {}).forEach(function (featureKey) {
```
Comment (mjc1283) [inline packages/optimizely-sdk/lib/core/project_config/index.js]:
```suggestion
    jsSdkUtils.objectValues(projectConfig.featureKeyMap || {}).forEach(function (feature) {
```

File: packages/optimizely-sdk/lib/optimizely/index.js [hunk 7]
```diff
@@ -629,14 +629,12 @@ Optimizely.prototype.getEnabledFeatures = function(userId, attributes) {
...
 
-    fns.forOwn(
-      configObj.featureKeyMap,
-      function(feature) {
-        if (this.isFeatureEnabled(feature.key, userId, attributes)) {
-          enabledFeatures.push(feature.key);
-        }
-      }.bind(this)
-    );
+    Object.keys(configObj.featureKeyMap).forEach(function (featureKey) {
```
Comment (mjc1283) [inline packages/optimizely-sdk/lib/optimizely/index.js]:
```suggestion
    jsSdkUtils.objectValues(configObj.featureKeyMap).forEach(function (feature) {
```

**Comment by mjc1283** [comment]
@zashraf1985 LGTM - could you please resolve the merge conflicts?

**Comment by zashraf1985** [comment]
@mjc1283 this is ready to be merged

**Pull request merged**